spark-statsd-reporter/src/main/java/org/apache/spark/metrics/sink/statsd/StatsdReporterFactory.java

```diff
@@ -0,0 +1,57 @@
+package org.apache.spark.metrics.sink.statsd;
+
+import com.codahale.metrics.MetricRegistry;
+
+import io.dropwizard.metrics.BaseReporterFactory;
+
+public class StatsdReporterFactory extends BaseReporterFactory {
+	private String host = "127.0.0.1";
+	private int port = 8125;
+	private String prefix = "";
+	private MetricFormatter metricFormatter;
+	
+	public String getHost() {
+        return host;
+    }
+
+    public void setHost(String host) {
+        this.host = host;
+    }
+    
+    public int getPort() {
+        return port;
+    }
+
+    public void setPort(int port) {
+        this.port = port;
+    }
+    
+    public String getPrefix() {
+        return prefix;
+    }
+
+    public void setPrefix(String prefix) {
+        this.prefix = prefix;
+    }
+    
+    public MetricFormatter getFormatter() {
+    	return metricFormatter;
+    }
+    
+    public void setFormatter(MetricFormatter metricFormatter) {
+    	this.metricFormatter = metricFormatter;
+    }
+
```

Comment:
It looks like all these setters overlap with the `Builder` in [StatsdReporter](https://github.com/mesosphere/spark-build/blob/master/spark-statsd-reporter/src/main/java/org/apache/spark/metrics/sink/statsd/StatsdReporter.java#L34-L82). Let's keep the code DRY and move the code from `Builder` here (including default values).

Reply:
If we move the code from `Builder` to `StatsdReporterFactory`, we have to remove the `Builder` from `StatsdReporter` and make constructor public. Will it be okay to do so?

Reply:
We can make `StatsdReporter` constructor package-private which is perfectly fine. As soon, as all modifications happen in a Builder, level of visibility is not that important - let's go forward with package-private visibility.